### PR TITLE
[SLS-1361] Use DD_XXX env vars for tagging

### PR DESCRIPTION
**Pull request by lizapressman**
### What does this PR do?

Allows for user to set tags for env, service, version, and tags under custom.datadog in their `serverless.yml` files that get translated to DD_XXX environment variables if using the extension and added as tags if using the forwarder.

If DD_XXX is specifically set at the provider or function level, that env var will not be overwritten. 

We also no longer set service and env tags if using the extension and not the forwarder. Instead these will be set as DD_ENV and DD_SERVICE environment variables that are still taken from the provider level (if not provided)

Also adds a warning message if a customer is using both forwarder and extension.

### Motivation

https://datadoghq.atlassian.net/browse/SLS-1361

### Testing Guidelines

Tested all the edge cases I could think of by deploying a [test python function](https://sa-east-1.console.aws.amazon.com/lambda/home?region=sa-east-1#/functions/liza-sls-testing-dev-hello_python36?tab=configure). Everything looked okay to me. 

### Additional Notes

It looks like support for DD_ENV, DD_SERVICE, and DD_VERSION were added in [v10](https://github.com/DataDog/datadog-lambda-extension/releases/tag/v10) of the extension and I believe DD_TAGS got support in [v9](https://github.com/DataDog/datadog-lambda-extension/releases/tag/v9) so anything using v9 or lower of the extension would break with these changes

### Types of changes

- [ ] Bug fix
- [x] New feature
- [x] Breaking change (For versions 9 or lower of the extension)
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [x] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

**Review by astuyve — APPROVED** [review]
Couple small nits. My only thought is, would this be considered a breaking change if paired with an older version of the extension which doesn't honor DD_* env variables? (IE: versions expecting tags to be crawled instead)?

If so, which versions? (so we can document it)

File: src/index.ts [hunk 1]
```diff
@@ -258,16 +276,98 @@ module.exports = class ServerlessPlugin {
...
+    let custom = service.custom as any;
+    if (custom === undefined) {
+      custom = {};
+    }
+
+    handlers.forEach(({ handler }) => {
+      handler.tags ??= {};
+      const tags = handler.tags as any;
+
+      if (custom?.datadog?.service !== undefined && tags[TagKeys.Service] === undefined) {
```
[the remark this answers is not in the document]
Comment (lizapressman) [reply src/index.ts:336]: I changed this and a lot of the other checks to 
```
if (custom?.datadog?.service) {
    tags[TagKeys.Service] ??= custom.datadog.service;
}
```
Is that cleaner or do you think sticking to 
```
if (custom?.datadog?.service && tags[TagKeys.Service] === undefined) {
    tags[TagKeys.Service] = custom.datadog.service;
}
```
is more readable/easier to understand what's happening?

[the remark this answers is not in the document]
Comment (astuyve) [reply src/index.ts]: I think the former is more readable, but I'm trying to consider a situation where `tags[TagKeys.Service]` would already be defined, and thus the behavior would change... Hmmm

[the remark this answers is not in the document]
Comment (lizapressman) [reply src/index.ts]: This was for the case of defining tags at the function level
```
functions:
  hello_python36:
    handler: handler.hello
    tags:
      service: function-level-test
```

**Comment by lizapressman** [comment]
> Couple small nits. My only thought is, would this be considered a breaking change if paired with an older version of the extension which doesn't honor DD_* env variables? (IE: versions expecting tags to be crawled instead)?
> 
> If so, which versions? (so we can document it)

It looks like support for DD_ENV, DD_SERVICE, and DD_VERSION were added in [v10](https://github.com/DataDog/datadog-lambda-extension/releases/tag/v10) of the extension and I believe DD_TAGS got support in [v9](https://github.com/DataDog/datadog-lambda-extension/releases/tag/v9) so I think anything using v9 or lower of the extension would break with these changes

**Comment by astuyve** [comment]
Okay! Let's add a note for that in the readme and then release this as a new major version.
